### ISSUE TITLE
Bug: Parsing errors in nested selectors

### ISSUE THREAD

**Issue by KristjanESPERANTO** [eslint/css#123]
### Environment

ESLint version: `9.26.0`
@eslint/css version: `0.7.0`
Node version: `23.6.1`
npm version: `10.9.0`
Operating System: `Debian 12`

### Which language are you using?

stylesheet

### What did you do?

<details>
<summary>Configuration</summary>

```js
import css from "@eslint/css";
import {defineConfig} from "eslint/config";

export default defineConfig([
  {files: ["**/*.css"], plugins: {css}, language: "css/css", extends: ["css/recommended"]},
]);

```
</details>

I get parsing errors in nested selectors - see comments in the following CSS example:

```css
.class1 {
  background-color: blue;

  span { /* <------------- Parsing error: Colon is expected */
    background-color: green;
  }
}

.class2 {
  background-color: yellow;

  div[class^="x"] { /* <------------- Parsing error: Colon is expected */
    background-color: red;
  }
}

.class3 {
  background-color: black;

  > thead { /* <------------- Parsing error: Identifier is expected */
    background-color: white;
  }
}
```

### What did you expect to happen?

No errors.

### What actually happened?

```bash
   4:8  error  Parsing error: Colon is expected
  12:6  error  Parsing error: Colon is expected
  20:3  error  Parsing error: Identifier is expected
```

### Link to Minimal Reproducible Example

-

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

-

**Comment by cbovis** [comment]
@KristjanESPERANTO tolerant parsing needs to be enabled for nested selectors.

https://github.com/eslint/css?tab=readme-ov-file#tolerant-mode

**Comment by Tanujkanti4441** [comment]
Yes, we just need to enable tolerant option here to avoid reporting parsing errors.

```js
languageOptions: {
     tolerant: true,
},
```

 [ASTExplorer](https://explorer.eslint.org/#eslint-explorer=IntcInN0YXRlXCI6e1widG9vbFwiOlwiYXN0XCIsXCJjb2RlXCI6e1wiamF2YXNjcmlwdFwiOlwiY2xhc3MgQSB7XFxuICAgICAgYSA9IDU7XFxuICAgICAgYiA9IHRoaXMuYTtcXG4gICAgICBjID0gdGhpcy5hO1xcbn1cIixcImpzb25cIjpcIi8qKlxcbiAqIFR5cGUgb3IgcGFzdGUgc29tZSBKU09OIGhlcmUgdG8gbGVhcm4gbW9yZSBhYm91dFxcbiAqIHRoZSBzdGF0aWMgYW5hbHlzaXMgdGhhdCBFU0xpbnQgY2FuIGRvIGZvciB5b3UuXFxuICpcXG4gKiBUaGUgdGFicyBhcmU6XFxuICpcXG4gKiAtIEFTVCAtIFRoZSBBYnN0cmFjdCBTeW50YXggVHJlZSBvZiB0aGUgY29kZSwgd2hpY2ggY2FuXFxuICogICBiZSB1c2VmdWwgdG8gdW5kZXJzdGFuZCB0aGUgc3RydWN0dXJlIG9mIHRoZSBjb2RlLiBZb3VcXG4gKiAgIGNhbiB2aWV3IHRoaXMgc3RydWN0dXJlIGFzIEpTT04gb3IgaW4gYSB0cmVlIGZvcm1hdC5cXG4gKlxcbiAqIFlvdSBjYW4gY2hhbmdlIHRoZSB3YXkgdGhhdCB0aGUgSlNPTiBjb2RlIGlzIGludGVycHJldGVkXFxuICogYnkgY2xpY2tpbmcgXFxcIkpTT05cXFwiIGluIHRoZSBoZWFkZXIgYW5kIHNlbGVjdGluZyBkaWZmZXJlbnRcXG4gKiBvcHRpb25zLlxcbiAqXFxuICogVGhpcyBleGFtcGxlIGlzIGluIEpTT05DIG1vZGUsIHdoaWNoIGFsbG93cyBjb21tZW50cy5cXG4gKi9cXG5cXG57XFxuICAgIFxcXCJrZXkxXFxcIjogW3RydWUsIGZhbHNlLCBudWxsXSxcXG4gICAgXFxcImtleTJcXFwiOiB7XFxuICAgICAgICBcXFwia2V5M1xcXCI6IFsxLCAyLCBcXFwiM1xcXCIsIDFlMTAsIDFlLTNdXFxuICAgIH1cXG59XCIsXCJtYXJrZG93blwiOlwiPCEtLVxcblR5cGUgb3IgcGFzdGUgc29tZSBNYXJrZG93biBoZXJlIHRvIGxlYXJuIG1vcmUgYWJvdXRcXG50aGUgc3RhdGljIGFuYWx5c2lzIHRoYXQgRVNMaW50IGNhbiBkbyBmb3IgeW91LlxcblxcblRoZSB0YWJzIGFyZTpcXG5cXG4tIEFTVCAtIFRoZSBBYnN0cmFjdCBTeW50YXggVHJlZSBvZiB0aGUgY29kZSwgd2hpY2ggY2FuXFxuYmUgdXNlZnVsIHRvIHVuZGVyc3RhbmQgdGhlIHN0cnVjdHVyZSBvZiB0aGUgY29kZS4gWW91XFxuY2FuIHZpZXcgdGhpcyBzdHJ1Y3R1cmUgYXMgSlNPTiBvciBpbiBhIHRyZWUgZm9ybWF0LlxcblxcbllvdSBjYW4gY2hhbmdlIHRoZSB3YXkgdGhhdCB0aGUgTWFya2Rvd24gY29kZSBpcyBpbnRlcnByZXRlZFxcbmJ5IGNsaWNraW5nIFxcXCJNYXJrZG93blxcXCIgaW4gdGhlIGhlYWRlciBhbmQgc2VsZWN0aW5nIGRpZmZlcmVudFxcbm9wdGlvbnMuXFxuXFxuVGhpcyBleGFtcGxlIGlzIGluIENvbW1vbk1hcmsgbW9kZS5cXG4tLT5cXG5cXG4jIEVTTGludCBNYXJrZG93biBFeGFtcGxlXFxuXFxuVGhpcyBpcyBhbiBleGFtcGxlIG9mIGEgTWFya2Rvd24gZmlsZSB0aGF0IGNhbiBiZSBwYXJzZWRcXG5ieSBFU0xpbnQuIE1hcmtkb3duIGlzIGEgc2ltcGxlIG1hcmt1cCBsYW5ndWFnZSB0aGF0IGlzXFxub2Z0ZW4gdXNlZCBmb3IgZG9jdW1lbnRhdGlvbi5cXG5cXG4jIyBGZWF0dXJlc1xcblxcbi0gTWFrZSB0aGluZ3MgKml0YWxpYyosICoqYm9sZCoqLCBvciBgY29kZWBcXG4tIENyZWF0ZSBbbGlua3NdKGh0dHBzOi8vZXNsaW50Lm9yZylcXG4tIFN1cHBvcnRzIEhUTUwgPHNwYW4gc3R5bGU9XFxcImNvbG9yOiByZWQ7XFxcIj5lbGVtZW50czwvc3Bhbj5cXG4tIExpc3RzXFxuICAtIE5lc3RlZCBsaXN0c1wiLFwiY3NzXCI6XCIuY2xhc3MxIHtcXG4gIGJhY2tncm91bmQtY29sb3I6IGJsdWU7XFxuXFxuICBzcGFuIHsgLyogPC0tLS0tLS0tLS0tLS0gUGFyc2luZyBlcnJvcjogQ29sb24gaXMgZXhwZWN0ZWQgKi9cXG4gICAgYmFja2dyb3VuZC1jb2xvcjogZ3JlZW47XFxuICB9XFxufVxcblxcbi5jbGFzczIge1xcbiAgYmFja2dyb3VuZC1jb2xvcjogeWVsbG93O1xcblxcbiAgZGl2W2NsYXNzXj1cXFwieFxcXCJdIHsgLyogPC0tLS0tLS0tLS0tLS0gUGFyc2luZyBlcnJvcjogQ29sb24gaXMgZXhwZWN0ZWQgKi9cXG4gICAgYmFja2dyb3VuZC1jb2xvcjogcmVkO1xcbiAgfVxcbn1cXG5cXG4uY2xhc3MzIHtcXG4gIGJhY2tncm91bmQtY29sb3I6IGJsYWNrO1xcblxcbiAgPiB0aGVhZCB7IC8qIDwtLS0tLS0tLS0tLS0tIFBhcnNpbmcgZXJyb3I6IElkZW50aWZpZXIgaXMgZXhwZWN0ZWQgKi9cXG4gICAgYmFja2dyb3VuZC1jb2xvcjogd2hpdGU7XFxuICB9XFxufVwifSxcImxhbmd1YWdlXCI6XCJjc3NcIixcImpzT3B0aW9uc1wiOntcInBhcnNlclwiOlwiZXNwcmVlXCIsXCJzb3VyY2VUeXBlXCI6XCJtb2R1bGVcIixcImVzVmVyc2lvblwiOlwibGF0ZXN0XCIsXCJpc0pTWFwiOnRydWV9LFwianNvbk9wdGlvbnNcIjp7XCJqc29uTW9kZVwiOlwianNvbmNcIn0sXCJjc3NPcHRpb25zXCI6e1wiY3NzTW9kZVwiOlwiY3NzXCIsXCJ0b2xlcmFudFwiOnRydWV9LFwibWFya2Rvd25PcHRpb25zXCI6e1wibWFya2Rvd25Nb2RlXCI6XCJjb21tb25tYXJrXCJ9LFwid3JhcFwiOnRydWUsXCJ2aWV3TW9kZXNcIjp7XCJhc3RWaWV3XCI6XCJ0cmVlXCIsXCJzY29wZVZpZXdcIjpcImZsYXRcIixcInBhdGhWaWV3XCI6XCJncmFwaFwifSxcInBhdGhJbmRleFwiOntcImluZGV4XCI6MSxcImluZGV4ZXNcIjozfSxcImVzcXVlcnlTZWxlY3RvclwiOntcInNlbGVjdG9yXCI6XCJcIn19LFwidmVyc2lvblwiOjB9Ig%3D%3D)

**Comment by KristjanESPERANTO** [comment]
Thank you both, that helped me! 🙂

But isn't the following sentence [in the docs](https://github.com/eslint/css#tolerant-mode) wrong?

`Setting tolerant to true is necessary if you are using custom syntax ... that aren't part of the standard CSS syntax.`

Nesting is now standard CSS syntax.

**Comment by nzakas** [comment]
Yes, nesting is part of CSS now and we are working on getting it implemented in the parser. It's not all there yet, so you can use `tolerant: true` to avoid parsing errors in the meantime.

**Comment by KristjanESPERANTO** [comment]
Many thanks for the explanation and also for all the work on ESLint! I am thrilled about the steps you have taken recently with ESLint.

**Comment by nzakas** [comment]
Thanks @KristjanESPERANTO for the feedback. We appreciate it! 🙏

**Comment by antti5** [comment]
I also ran into this issue with certain nested CSS selectors, and I was about to create a new issue. However, I decided double-check the closed issues, and I found that the tolerant mode indeed solved also my issue.

Maybe add a mention of this to the NPM page? Tolerant mode is described, but there's no mention that it could be necessary for perfectly standard CSS syntax.

One of the selectors that caused parsing errors for me is what I think is a fairly common nested selector: `a:focus &`

**Comment by nzakas** [comment]
@antti5 please open a separate issue with a code example that is still failing for you.
